### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by bmarty** [vector-im/element-android#3605]
```
Thread: main, Exception: java.lang.IllegalArgumentException: Configurations cannot be different if used to open the same file. The most likely cause is that equals() and hashCode() are not overridden in the migration class: org.matrix.android.sdk.internal.database.RealmSessionStoreMigration
at io.realm.RealmCache.validateConfiguration(RealmCache.java:10)
at io.realm.Realm.getInstance(Realm.java:20)
at org.matrix.android.sdk.internal.session.SessionModule_ProvidesRealmConfigurationFactory.get(SessionModule_ProvidesRealmConfigurationFactory.java:36)
at dagger.internal.DoubleCheck.get(DoubleCheck.java:5)
at org.matrix.android.sdk.internal.session.DefaultSession_Factory.get(DefaultSession_Factory.java:1)
at dagger.internal.DoubleCheck.get(DoubleCheck.java:5)
at org.matrix.android.sdk.internal.session.DaggerSessionComponent.session(DaggerSessionComponent.java:1)
at org.matrix.android.sdk.internal.session.SessionListeners.dispatch(SessionListeners.kt:2)
at org.matrix.android.sdk.internal.session.-$$Lambda$DefaultSession$9D4dsVdsKGBrgc18THUFICCTKcQ.run(lambda:5)
at android.os.Handler.handleCallback(Handler.java:938)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:239)
at android.app.ActivityThread.main(ActivityThread.java:8142)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:626)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1015)
```

**Comment by bmarty** [comment]
Use object instead of class for Realm Migration. Pb with RealmCryptoStoreMigration, it's injected with an non-empty constructor.

**Comment by bmarty** [comment]
Included in 1.1.12
